public/src/main/java/com/revenuecat/purchases/DangerousSettings.kt

```diff
@@ -9,5 +9,11 @@ data class DangerousSettings(
      * automatically, and you will have to call syncPurchases whenever a new purchase is completed in order to send it
      * to the RevenueCat's backend. Auto syncing of purchases is enabled by default.
      */
-    val autoSyncPurchases: Boolean = true
+    val autoSyncPurchases: Boolean = true,
+
+    /**
+     * Disable or enable to force and test a situation where the RevenueCat server is down. This is meant to only be
+     * enabled during tests.
+     */
+    val forceServerDown: Boolean = false
```

Comment:
this is a public-facing setting, right? I mean... can we _proxy_ that setting through something else? 
For example, just mock http calls and return an error, or use the `proxyURL` setting, using a value that we know will return a 5xx?

Reply:
I see your point... I am not sure about making this public either. @NachoSoto proposed the same for iOS, but maybe it's not that good of an idea after all. This can easily be achieved with Charles, so maybe we should just not merge this

Reply:
on iOS it's internal, though, so it's not much of a problem

Reply:
Hmm if it's internal I guess purchase-tester won't be able to access it though right? But yeah, I guess we shouldn't expose this...

Reply:
we can kind of cheat on iOS with `@testable import`, but we can't do that on Android. 
Honestly, I'm good with any solution as long as it doesn't involve exposing APIs that we don't need a customer to ever use. 
In particular, for this one all we need is to figure out a way to have an API that returns 5xx. It feels like we should be able to accomplish that without much work

Reply:
like, here's chatgpt's suggestion for doing a server in ruby that returns 5xx: 
<img width="667" alt="image" src="https://user-images.githubusercontent.com/3922667/228633147-f0d0c23b-c76a-4c22-8500-b403ae13b13e.png">
We can do that, then for integration tests, allow cleartest traffic, set proxyURL to localhost:8080, call it a day